pkg/reconciler/revision/resources/pa.go

```diff
@@ -44,27 +45,30 @@ func MakePA(rev *v1.Revision) *autoscalingv1alpha1.PodAutoscaler {
 				Name:       names.Deployment(rev),
 			},
 			ProtocolType: rev.GetProtocol(),
-			Reachability: func() autoscalingv1alpha1.ReachabilityType {
-				// If the Revision has failed to become Ready, then mark the PodAutoscaler as unreachable.
-				if rev.Status.GetCondition(v1.RevisionConditionReady).IsFalse() {
-					// Make sure that we don't do this when a newly failing revision is
-					// marked reachable by outside forces.
-					if !rev.IsReachable() {
-						return autoscalingv1alpha1.ReachabilityUnreachable
-					}
-				}
+			Reachability: reachability(rev),
+		},
+	}
+}
 
-				// We don't know the reachability if the revision has just been created
-				// or it is activating.
-				if rev.Status.GetCondition(v1.RevisionConditionActive).IsUnknown() {
-					return autoscalingv1alpha1.ReachabilityUnknown
-				}
+func reachability(rev *v1.Revision) autoscalingv1alpha1.ReachabilityType {
+	// check infra failures
+	conds := []apis.ConditionType{
+		v1.RevisionConditionResourcesAvailable,
+		v1.RevisionConditionContainerHealthy,
+	}
 
-				if rev.IsReachable() {
```

Comment:
Since `IsReachable()` util func is no longer used anywhere, shall we remove it from https://github.com/dprotaso/serving/blob/reachability-fix/pkg/apis/serving/v1/revision_helpers.go#L131-L134 ?

Reply:
I'm ok leaving it - unsure if anyone else is using it downstream

Reply:
dropped it